File: Data Structures & Algorithms/palindromic-substrings/submission-3.cpp
class Solution {
public:
int count(string s,int left,int right){
    int ans=0;
    while(left>=0 && right<s.size() && s[left]==s[right]){
        left--;
        right++;
        ans++;
    }
    return ans;
}
    int countSubstrings(string s) {
        int res=0;
        for(int i=0;i<s.size();i++){
            res+=count(s,i,i);
            res+=count(s,i,i+1);
        }
        return res;
    }
};

File: Data Structures & Algorithms/longest-palindromic-substring/submission-3.java
class Solution {
    String solve(int left,int right,String s){
        while(left>=0 && right<s.length() && s.charAt(left)==s.charAt(right)){
            left--;
            right++;
        }
        return s.substring(left+1,right);
    }
    public String longestPalindrome(String s) {
        int n=s.length();
        String ans="";
        for(int i=0;i<n;i++){
            String even=solve(i,i,s);
            String odd=solve(i,i+1,s);
            if(even.length()>ans.length()) ans=even;
            if(odd.length()>ans.length()) ans=odd;
        }
        return ans;
    }
}
